src/integrators/sdirk_integrators.jl

```diff
@@ -481,204 +465,211 @@ end
   f(t+dt,u,integrator.fsallast)
 end
 
-function initialize!(integrator,cache::TRBDF2ConstantCache,f=integrator.f)
+function initialize!(integrator, cache::TRBDF2ConstantCache)
   integrator.kshortsize = 2
   integrator.k = eltype(integrator.sol.k)(integrator.kshortsize)
-  integrator.fsalfirst = f(integrator.t,integrator.uprev) # Pre-start fsal
+  integrator.fsalfirst = integrator.f(integrator.t, integrator.uprev) # Pre-start fsal
 
   # Avoid undefined entries if k is an array of arrays
   integrator.fsallast = zero(integrator.fsalfirst)
   integrator.k[1] = integrator.fsalfirst
   integrator.k[2] = integrator.fsallast
 end
 
-@muladd function perform_step!(integrator,cache::TRBDF2ConstantCache,f=integrator.f)
-  @unpack t,dt,uprev,u = integrator
-  @unpack uf = cache
-  uf.t = t
+@muladd function perform_step!(integrator, cache::TRBDF2ConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f = integrator
+  @unpack uf,κ,tol = cache
+
+  # Precalculations
+  # -> move constants to cache?
   γ = 2 - sqrt(2)
   ω = sqrt(2)/4
   d = γ/2
 
-  b1 = ω
-  bhat1 = (1-ω)/3
-  b2 = ω
-  bhat2 = (3ω + 1)/3
-  b3 = d
-  bhat3 = d/3
+  a1 = -sqrt(2)/2
+  a2 = 1 + sqrt(2)/2
+
+  c1 = γ/4
+  c2 = a2/2
+
+  btilde1 = (1-sqrt(2))/3
+  btilde2 = 1/3
+  btilde3 = (sqrt(2)-2)/3
+
+  κtol = κ*tol # save in cache instead of κ and tol?
 
   γdt = γ*dt
-  κ = cache.κ
-  tol = cache.tol
+  ddt = d*dt
 
+  uf.t = t
   if typeof(uprev) <: AbstractArray
     J = ForwardDiff.jacobian(uf,uprev)
-    W = I - d*dt*J
+    W = I - ddt*J
   else
     J = ForwardDiff.derivative(uf,uprev)
-    W = 1 - d*dt*J
+    W = 1 - ddt*J
   end
 
   # TODO: Add extrapolant initial guess
-  zprev = dt*integrator.fsalfirst
+  zprev = dt.*integrator.fsalfirst
 
   ##### Solve Trapezoid Step
 
   zᵧ = zprev
```

Comment:
As with below, the extrapolant comment should be moved down here. And it shouldn't be assumed this will hold.